client-v2/src/components/util/WithDimensions.tsx

```diff
@@ -24,28 +23,26 @@ const initialState = { width: undefined, height: undefined } as Dimensions;
  * </WithDimensions>
  * ```
  */
-const WithDimensions = ({ element, children }: Props) => {
+const WithDimensions = ({ children }: Props) => {
   const [dimensions, setDimensions] = useState(initialState);
-  useEffect(
-    () => {
-      if (!element) {
+  const ref: React.RefObject<HTMLDivElement> = React.createRef();
+  useLayoutEffect(() => {
+    if (!ref.current) {
+      return;
+    }
+    const observer = new ResizeObserver(([firstEntry]) => {
+      if (!firstEntry) {
         return;
       }
-      const observer = new ResizeObserver(([firstEntry]) => {
-        if (!firstEntry) {
-          return;
-        }
-        const { width, height } = firstEntry.contentRect;
-        setDimensions({ width, height });
-      });
+      const { width, height } = firstEntry.contentRect;
+      setDimensions({ width, height });
+    });
 
-      observer.observe(element);
+    observer.observe(ref.current);
 
-      return () => observer.disconnect();
-    },
-    [element]
-  );
-  return children(dimensions);
+    return () => observer.disconnect();
+  }, []);
+  return <div ref={ref}>{children(dimensions)}</div>;
```

Comment:
I've reluctantly added a container div here, as without a ref that this component owns, `useLayout` can't guarantee that refs of other components are present in the dom. This was effectively working by coincidence before this point, due to redundant rerenders.